### ISSUE TITLE
question on L2 loss

### ISSUE THREAD

**Issue by adoskk** [kctezcan/ddp_recon#1]
Hi,

I have a naive question about the l2 loss constructed here: https://github.com/kctezcan/ddp_recon/blob/736bc6f536858a08c4228fe977ebd00570ccd8df/definevae.py#L175 

My interpretation is that the assumption of the likelihood is Gaussian, y_out is the mean and y_out_prec is the variance. In that sense, the l2_loss_1 should be tf.reduce_sum(tf.divide(tf.pow((y_out - x_inp__),2), y_out_prec),axis=1) instead, not using tf.multiply. But I might be totally wrong. Your clarification and help would really be appreciated!

**Comment by kctezcan** [comment]
Dear Megliu Zhao,
I think the y_out_prec predicts the precision vale, i.e. the inverse of the
variance. Hence the multiplication instead of the division.
That means the sign of l2_loss_2 has to be used accordingly.
Best,
Kerem

Mengliu Zhao ***@***.***> schrieb am So. 14. Nov. 2021 um
23:38:
